packages/status-js/src/request-client/request-client.ts

```diff
@@ -127,57 +143,80 @@ class RequestClient {
-  private fetchCommunityDescription = async (
-    /** Uncompressed */
-    publicKey: string
-  ): Promise<CommunityDescription | undefined> => {
-    const contentTopic = idToContentTopic(publicKey)
-    const symmetricKey = await generateKeyFromPassword(publicKey)
+  public fetchCommunityDescription = async (
+    /** Compressed */
+    publicKeys: string[]
+  ): Promise<Map<string, CommunityDescription>> => {
+    const decoderPromises = publicKeys.map(async publicKey => {
+      const contentTopic = idToContentTopic(publicKey)
+      const symmetricKey = await generateKeyFromPassword(publicKey)
 
-    let communityDescription: CommunityDescription | undefined = undefined
-    await this.waku.store.queryOrderedCallback(
-      [new SymDecoder(contentTopic, symmetricKey)],
-      wakuMessage => {
-        // handle
-        const message = this.handleWakuMessage(wakuMessage)
+      return new SymDecoder(contentTopic, symmetricKey)
+    })
 
-        if (!message) {
-          return
-        }
+    const decoders = await Promise.all(decoderPromises)
 
-        if (
-          message.type !== ApplicationMetadataMessage_Type.COMMUNITY_DESCRIPTION
-        ) {
-          return
-        }
+    const communityDescription: Map<string, CommunityDescription> = new Map()
+    await this.waku.store.queryOrderedCallback(decoders, wakuMessage => {
+      // handle
+      const message = this.handleWakuMessage(wakuMessage)
 
-        // decode
-        const decodedCommunityDescription = CommunityDescription.fromBinary(
-          message.payload
+      if (!message) {
+        return
+      }
+
+      if (
+        message.type !== ApplicationMetadataMessage_Type.COMMUNITY_DESCRIPTION
+      ) {
+        return
+      }
+
+      // decode
+      const decodedCommunityDescription = CommunityDescription.fromBinary(
+        message.payload
+      )
+
+      // validate
+      if (
+        !isClockValid(
+          BigInt(decodedCommunityDescription.clock),
+          message.timestamp
         )
+      ) {
+        return
+      }
 
-        // validate
-        if (
-          !isClockValid(
-            BigInt(decodedCommunityDescription.clock),
-            message.timestamp
-          )
-        ) {
-          return
-        }
+      console.log(decodedCommunityDescription)
 
-        if (publicKey !== `0x${compressPublicKey(message.signerPublicKey)}`) {
-          return
-        }
+      const decodedCommunityPublicKey = `0x${compressPublicKey(
+        message.signerPublicKey
+      )}`
 
-        if (!communityDescription) {
-          communityDescription = decodedCommunityDescription
-        }
+      if (!publicKeys.includes(decodedCommunityPublicKey)) {
+        return
+      }
 
-        // stop
-        return true
+      if (!communityDescription.has(decodedCommunityPublicKey)) {
+        // todo?: ensure mapping back to original pk format if deserialized in this fn
+        communityDescription.set(
+          decodedCommunityPublicKey,
```

Comment:
Expects caller to handle compressed pks or keep a map of the original format and deserialized one.